spyder_line_profiler/spyder/plugin.py

```diff
@@ -0,0 +1,128 @@
+
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtCore import Signal
+from qtpy.QtGui import QIcon
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+from spyder.api.plugin_registration.decorators import on_plugin_teardown
+from spyder.plugins.mainmenu.api import ApplicationMenus
+
+# Local imports
+from spyder_line_profiler.spyder.confpage import SpyderLineProfilerConfigPage
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidget
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidgetActions
+from spyder_line_profiler.spyder.widgets import is_lineprofiler_installed
+
+# Localization
+_ = get_translation("spyder_line_profiler.spyder")
+
+
+
+class SpyderLineProfilerActions:
+    # Triggers
+    Run = 'run_profiler_action'
+    
+
+class SpyderLineProfiler(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler plugin for Spyder 5.
+    """
+
+    NAME = "spyder_line_profiler"
+    REQUIRES = [Plugins.Editor]
+    OPTIONAL = [Plugins.MainMenu]
+    TABIFY = [Plugins.Help]
+    WIDGET_CLASS = SpyderLineProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfilerConfigPage
+
+    # --- Signals
+    sig_finished = Signal()
+    """This signal is emitted to inform the profile profiling has finished."""
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Line Profiler")
+
+    def get_description(self):
+        return _("Line profiler display for Spyder")
+
+    def get_icon(self):
+        return QIcon('./data/images/spyder.line_profiler.png')
+
+    def on_initialize(self):
+        self.widget = self.get_widget()
+        self.widget.sig_finished.connect(self.sig_finished)
+        
+        run_action = self.create_action(
+            SpyderLineProfilerActions.Run,
+            text=_("Run line profiler"),
+            tip=_("Run line profiler"),
+            icon=self.get_icon(),
+            triggered=self.run_lineprofiler,
+            register_shortcut=True,
+        )   
+        run_action.setEnabled(is_lineprofiler_installed())
+        
+    @on_plugin_available(plugin=Plugins.MainMenu)
+    def on_main_menu_available(self):
+        mainmenu = self.get_plugin(Plugins.MainMenu)
+        run_action = self.get_action(SpyderLineProfilerActions.Run)
+        mainmenu.add_item_to_application_menu(
+            run_action, menu_id=ApplicationMenus.Run)
+        
+    @on_plugin_teardown(plugin=Plugins.MainMenu)
+    def on_main_menu_teardown(self):
+        mainmenu = self.get_plugin(Plugins.MainMenu)
+        mainmenu.remove_item_from_application_menu(
+            SpyderLineProfilerActions.ProfileCurrentFile,
+            menu_id=ApplicationMenus.Run
+        )    
+        
```

Comment:
Missing plugin Preferences registration:
```suggestion

    @on_plugin_available(plugin=Plugins.Preferences)
    def on_preferences_available(self):
        preferences = self.get_plugin(Plugins.Preferences)
        preferences.register_plugin_preferences(self)

```